### PR TITLE
Document that `--diff` implies `--fix-only`

### PR DESCRIPTION
**Pull request by dhruvmanila**
fixes: #4093

### CODE REVIEW

**Comment by JonathanPlasse** [comment]
It could also work if fix is set in the toml and not the CLI.

**Comment by MichaReiser** [comment]
Thanks for looking into this issue.

If I understand this correctly, then there's no use case for using `--diff` without also using `--fix`. Do I understand this correctly? 

If so, wouldn't it be easier for user if we change the semantics of `--diff` to either automatically imply `--fix` and/or rename the option to e.g. `--fix-dry-run` (similar to [ESLint](https://eslint.org/docs/latest/use/command-line-interface#--fix-dry-run)) to make it clear that it computes the fixes?

**Comment by charliermarsh** [comment]
Hm yeah, at the very least, I think `--diff` should imply `--fix` or perhaps `--fix-only`.

I did look at `--fix-dry-run` at the time, but I think I opted for `--diff` to match Black's `--diff` option.

**Comment by dhruvmanila** [comment]
> It could also work if fix is set in the toml and not the CLI.

I don't understand this as there's no `diff` option in configuration. This is (or was) about the CLI _only_ flag `--diff` 

---

> If I understand this correctly, then there's no use case for using `--diff` without also using `--fix`. Do I understand this correctly?

Yes

> If so, wouldn't it be easier for user if we change the semantics of `--diff` to either automatically imply `--fix` and/or rename the option to e.g. `--fix-dry-run` (similar to [ESLint](https://eslint.org/docs/latest/use/command-line-interface#--fix-dry-run)) to make it clear that it computes the fixes?

There's a `TODO` comment to enable it, but I think it's out of scope for this PR.

https://github.com/charliermarsh/ruff/blob/bff5d98121394c93bc9f82c45285517f85f71f41/crates/ruff_cli/src/lib.rs#L134-L136

---

> Hm yeah, at the very least, I think `--diff` should imply `--fix` or perhaps `--fix-only`.

Yes, I think it should imply `--fix-only` as it's to only show the fix diff and no violations are reported.

---

**tl;dr**, this should basically just be a document change to specify that `--diff` option implies `--fix-only`

**Comment by charliermarsh** [comment]
Ahh right, sorry, you're correct. `--diff` _does_ imply `--fix-only`, it's just not documented.
